prov/efa/src/rdm/efa_rdm_cq.c

```diff
@@ -373,7 +373,10 @@ static void efa_rdm_cq_handle_recv_completion(struct efa_ibv_cq *ibv_cq, struct
 	if (OFI_UNLIKELY(ep->use_zcpy_rx && efa_rdm_pkt_type_is_rtm(pkt_type))) {
 		EFA_WARN(FI_LOG_CQ,
 			"Invalid pkt type %d! Peer %d doesn't respect the request from this EP that"
-			" RTM packets must be sent to the user recv QP.\n",
+			" RTM packets must be sent to the user recv QP. One possible reason is that"
+			" the sender is using an old libfabric that doesn't support the user recv QP feature."
+			" Please consider upgrading the sender's libfabric version to >= 1.22.0, or disable"
+			" the zero-copy recv feature by setting FI_EFA_USE_ZCPY_RX=0 in the environment",
```

Comment:
I think we can be more concise with the error message, e.g.

```
Packet type <type> from remote peer <instance-id>/<gid> is not supported when zero-copy recv mode is ON.
 Please ensure local and remote libfabric versions match, or turn off zero-copy recv via
 FI_EFA_USE_ZCPY_RX=0.
```

Reply:
We should relegate these long, passage-like messages to `efa_show_help()`

Reply:
If we use `efa_show_help()`, I am not sure what error code should I populate here? It seems a very specific corner case instead of a general libfabric error

Reply:
>Please ensure local and remote libfabric versions match

Libfabric was supposed to be backward compatibility (with some corner cases for sure, this is one of them). I do not think we should enforce users to have matched libfabric versions on local and remote

Reply:
> I am not sure what error code should I populate here? It seems a very specific corner case instead of a general libfabric error

I'd suggest a new error code; similar to what we did for unresponsive receiver with/without a successful handshake.

If it can't really be detected in a similar way, it'd make a good backlog item.

Reply:
Done, see latest revision.